scripts/mypyc

```diff
@@ -27,29 +30,128 @@ from mypyc.build import mypycify
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('subcommand', type=str, choices=['build', 'run', 'clean'],
+                        help='mypyc subcommand')
+    parser.add_argument(metavar='files', nargs='*', dest='files',
+                        help="Compile given files")
+    args = parser.parse_args()
+    options.mode = mode_mapping[args.subcommand]
+    options.files = args.files
+    return options
+
+
+def mypyc_compile(build_dir: str,
+                  paths: List[str],
+                  opt_level: str) -> int:
     try:
         os.mkdir(build_dir)
     except FileExistsError:
         pass
-
-    opt_level = os.getenv("MYPYC_OPT_LEVEL", '3')
-
     setup_file = os.path.join(build_dir, 'setup.py')
     with open(setup_file, 'w') as f:
-        f.write(setup_format.format(sys.argv[1:], opt_level))
-
+        f.write(setup_format.format(paths, opt_level))
     # We don't use run_setup (like we do in the test suite) because it throws
     # away the error code from distutils, and we don't care about the slight
     # performance loss here.
     env = os.environ.copy()
     base_path = os.path.join(os.path.dirname(__file__), '..')
     env['PYTHONPATH'] = base_path + os.pathsep + env.get('PYTHONPATH', '')
     cmd = subprocess.run([sys.executable, setup_file, 'build_ext', '--inplace'], env=env)
+    return cmd.returncode
+
+
+def mypyc_build(options: Options) -> None:
+    if len(options.files) < 1:
+        sys.exit("no source files provided")
+    returncode = mypyc_compile(options.build_dir, options.files, options.opt_level)
+    sys.exit(returncode)
+
+
+def mypyc_run(options: Options) -> None:
+    if len(options.files) < 1:
+        sys.exit("no source file provided")
+    returncode = mypyc_compile(options.build_dir, options.files[0:1], options.opt_level)
+    if returncode != 0:
+        sys.exit(returncode)
+    module_name = os.path.basename(options.files[0])
+    module_name = os.path.splitext(module_name)[0]
+    import_command = '"import {}"'.format(module_name)
+    # TODO: well this is dumb, we'd come up with a better way
+    env = os.environ.copy()
+    base_path = os.path.join(os.path.dirname(__file__), '..')
+    env['PYTHONPATH'] = base_path + os.pathsep + env.get('PYTHONPATH', '')
+    # TODO: How do we guarantee that we run the compiled version when both .py and .so files
+    # are in the same directory?
+    cmd = subprocess.run("python -c " + import_command, shell=True)
+    # TODO: I don't know why this is not working...
+    # cmd = subprocess.run([sys.executable, 'cmd', import_command], env=env, capture_output=True)
     sys.exit(cmd.returncode)
 
+
+def mypyc_clean(options: Options) -> None:
+    if os.path.exists(options.build_dir):
+        files = []
+        directories = []
+        # TODO: this simply hardcodes what generates now
+        # a better solution is to use some configuration files
+        # maybe use a `.mypyc_cache` folder to store them
```

Comment:
Brainstorming: maybe the `build` directory should be renamed to something with `mypyc` in the name, perhaps `__mypyc__` or something, and then we can assume that everything in the directory are owned by us can be deleted.

Reply:
`__mypyc__` would be a decent choice, but I am worried about the generated shared object files? Currently, we place them under cwd. To clean them, we probably need to search all `*.so` in cwd and delete them, which IMO is not a good approach.

Reply:
Deleting the `.so` files would make the command much more useful, I think. 

We should be able to maintain the paths to all generated `.so` files in a file under `build`/`__mypyc__` and delete only the files we have generated. If we are still worried that we may deleting the wrong files, we can also record the mtime and size of the files, and only delete files if the mtime and size match what we have recorded.

Reply:
This approach looks way better. Let's try go with `__mypyc__` to place all the `*.c, *.h, ops.txt, setup.py`(or any other intermediate scripts) and `__mypyc__/.mypyc_meta`(the name can be tweak) to record such file information.